modules/account-lib/src/coin/dot/transaction.ts

```diff
@@ -1,20 +1,211 @@
-import { BaseTransaction } from '../baseCoin';
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { BaseTransaction, TransactionType } from '../baseCoin';
 import { BaseKey } from '../baseCoin/iface';
-import { NotImplementedError } from '../baseCoin/errors';
+import { InvalidTransactionError, SigningError } from '../baseCoin/errors';
+import { construct, decode } from '@substrate/txwrapper-polkadot';
+import { UnsignedTransaction } from '@substrate/txwrapper-core';
+import { TypeRegistry } from '@substrate/txwrapper-core/lib/types';
+import { decodeAddress } from '@polkadot/keyring';
+import { KeyPair } from './keyPair';
+import {
+  TxData,
+  DecodedTx,
+  TransferArgs,
+  StakeArgs,
+  StakeArgsPayeeRaw,
+  AddProxyArgs,
+  ProxyArgs,
+  UnstakeArgs,
+} from './iface';
+import utils from './utils';
 
 export class Transaction extends BaseTransaction {
+  protected _dotTransaction: UnsignedTransaction;
+  private _signedTransaction?: string;
+  private _registry: TypeRegistry;
+  private _chainName: string;
+  private _sender: string;
+
+  constructor(coinConfig: Readonly<CoinConfig>) {
+    super(coinConfig);
+  }
+
   /** @inheritdoc */
-  canSign(key: BaseKey): boolean {
-    throw new NotImplementedError('canSign not implemented');
+  canSign({ key }: BaseKey): boolean {
+    const kp = new KeyPair({ prv: key });
+    const addr = kp.getAddress();
+    if (addr === this._sender) {
+      return true;
+    }
+    return false;
+  }
+
+  async sign(keyPair: KeyPair): Promise<void> {
+    if (!this._dotTransaction) {
+      throw new InvalidTransactionError('No transaction data to sign');
+    }
+    if (!keyPair.getKeys().prv) {
+      throw new SigningError('Missing private key');
+    }
+    const signingPayload = construct.signingPayload(this._dotTransaction, {
+      registry: this._registry,
+    });
+    // Sign a payload. This operation should be performed on an offline device.
+    const signingKeyPair = keyPair.getSigningKeyPair();
+    const txHex = utils.createSignedTx(signingKeyPair, signingPayload, this._dotTransaction, {
+      metadataRpc: this._dotTransaction.metadataRpc,
+      registry: this._registry,
+    });
+
+    this._signedTransaction = txHex;
+  }
+
+  registry(registry: TypeRegistry): void {
+    this._registry = registry;
+  }
+
+  chainName(chainName: string): void {
```

Comment:
Is this the network? Do we need this if we have the network name in statics?

Reply:
yes, letting this setter be dynamic